src/lib.rs

```diff
@@ -313,8 +317,8 @@ impl OctocrabBuilder {
     }
 
     /// Set the base url for `Octocrab`.
-    pub fn base_url(mut self, base_url: impl reqwest::IntoUrl) -> Result<Self> {
-        self.base_url = Some(base_url.into_url().context(crate::error::HttpSnafu)?);
+    pub fn base_url(mut self, base_url: &str) -> Result<Self> {
```

Comment:
This should be `AsRef<str>`, same with the other methods.

Reply:
Done! I am fairly new to rust so I might have these amateurish errors 🥲.

Reply:
Everyone has to learn sometime, don't be afraid to ask questions in the PR if you're unsure about something 🙂